x-pack/scripts/functional_tests.js

```diff
@@ -52,6 +52,7 @@ const onlyNotInCoverageTests = [
   require.resolve('../test/endpoint_api_integration_no_ingest/config.ts'),
   require.resolve('../test/reporting_api_integration/config.js'),
   require.resolve('../test/functional_embedded/config.ts'),
+  require.resolve('../test/ingest_manager_api_integration'),
```

Comment:
Thanks for working on this Sonja! The endpoint team will also hopefully be able to pick up the docker registry changes in our tests too.

How is the package storage repo brought in with the registry? Or do we have to commit the packages to kibana that we want the registry to serve for the tests?

Reply:
Right now the docker container comes with a certain set of packages. I'm using a container from a specific PR (see here: https://github.com/elastic/kibana/pull/69822/files#diff-61a40c93b50ddc25b22ebb367b272aa9R21 ) but it's not very easy to find out what packages and versions are actually included in that build. I don't have a solution for that yet and am open to all suggestions.

In addition to that it is possible to mount a directory into the container and tell the package registry to serve packages from that in addition to those that are delivered with the container, so it will be possible to craft special packages to test specific corner cases. (Watch out for the next commits on this PR how this will work.)

The issue with that approach is that our production packages contain lots of small files, and I'm not comfortable with adding them to the test directory. If nothing else, it makes PRs unreadable (see https://github.com/spalger/kibana/pull/34 for an example).

Reply:
> Right now the docker container comes with a certain set of packages. I'm using a container from a specific PR

Are we able to override that package registry version when running the tests locally? If I understand correctly we can just set `INGEST_MANAGEMENT_PACKAGE_REGISTRY_PORT` to the port of our locally running registry?

> The issue with that approach is that our production packages contain lots of small files, and I'm not comfortable with adding them to the test directory. If nothing else, it makes PRs unreadable (see spalger#34 for an example).

Yeah I noticed that too. @ruflin I wonder if we could add functionality to the registry so it could decompress a tar.gz file to get extra packages. That way we could at least commit gzipped files here for tests. It still doesn't make it clear what packages are included until someone decompresses it but it makes it easier to review.

Another option is we could pull down a specific branch of the package storage repo but adding another external dependency might not be a good idea. Plus that might conflict with the proposed package release flow.

Also can we pin the registry to specific commits or is it just `master` and specific PRs?

Reply:
> Also can we pin the registry to specific commits or is it just master and specific PRs?

You can use anything that is available on https://container-library.elastic.co/r/package-registry/package-registry

> @ruflin I wonder if we could add functionality to the registry so it could decompress a tar.gz file to get extra packages. That way we could at least commit gzipped files here for tests.

Seconded, that would make it easier to add extra packages for tests.

Reply:
> Are we able to override that package registry version when running the tests locally?

The docker container that is configured in the `config.{ts,js}` for your `*_api_integration` directory is used for all the tests. It is currently not possible to start up a docker container per test. That means that all packages you need in addition to those in the container will need to be in one central place for all your tests.

> If I understand correctly we can just set INGEST_MANAGEMENT_PACKAGE_REGISTRY_PORT to the port of our locally running registry?

No. This is an environment variable that will be set by CI tooling. The tests are run in a distributed way, and it is possible that one jenkins worker runs two sets of tests using docker at the same time. Therefore, all the ports that docker containers are mounted to need to be unter control of the CI environment to ensure there are no port conflicts.

When the tests are run locally, the presence of this environment variable indicates that the tests should be run at all. Leaving it out makes it possible to run the tests on a machine without docker. This all will be documented.

Reply:
Sounds good, thanks for the explanation!

Reply:
I think adding files instead of tar.gz is a feature. If a package is changed for a test, the PR reviewer should also be able to review the package change. If it is in a .tar.gz, the user can't. 

My expectation is that for the tests, we build very small packages dedicated to what we want to test. So the number of files will be low and once commited, from the on the diffs will only be on the changes. I don't expect us to push a new package version with all the files each time we update a package here.

If we want later to test across many packages I would rather have a testing distribution of the registry with these packages already inside instead of storing all the .tar.gz inside, but I think this is a different testing scenario.